File: guides/java/kora-java-guide-database-jdbc-advanced-app/src/main/java/io/koraframework/guide/databasejdbc/advanced/task/repository/mapper/ListOfLongJdbcParameterMapper.java
package io.koraframework.guide.databasejdbc.advanced.task.repository.mapper;

import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.sql.Types;
import java.util.List;
import io.koraframework.common.annotation.Component;
import io.koraframework.database.jdbc.mapper.parameter.JdbcParameterColumnMapper;

@Component
public final class ListOfLongJdbcParameterMapper implements JdbcParameterColumnMapper<List<Long>> {

    @Override
    public void set(PreparedStatement stmt, int index, List<Long> value) throws SQLException {
        if (value == null) {
            stmt.setNull(index, Types.ARRAY);
            return;
        }

        var sqlArray = stmt.getConnection().createArrayOf("BIGINT", value.toArray(Long[]::new));
        stmt.setArray(index, sqlArray);
    }
}

File: guides/kotlin/kora-kotlin-guide-database-jdbc-advanced-app/src/main/kotlin/io/koraframework/guide/databasejdbc/advanced/task/repository/mapper/ListOfLongJdbcParameterMapper.kt
package io.koraframework.guide.databasejdbc.advanced.task.repository.mapper

import io.koraframework.common.annotation.Component
import io.koraframework.database.jdbc.mapper.parameter.JdbcParameterColumnMapper
import java.sql.PreparedStatement
import java.sql.Types

@Component
class ListOfLongJdbcParameterMapper : JdbcParameterColumnMapper<List<Long>> {

    // the 2.0 contract declares the value as @Nullable, which Kotlin enforces on the override
    override fun set(stmt: PreparedStatement, index: Int, value: List<Long>?) {
        if (value == null) {
            stmt.setNull(index, Types.ARRAY)
            return
        }

        val sqlArray = stmt.connection.createArrayOf("BIGINT", value.toTypedArray())
        stmt.setArray(index, sqlArray)
    }
}
